crates/bevy_ecs/src/query/access.rs

```diff
@@ -0,0 +1,199 @@
+use crate::storage::SparseSetIndex;
+use fixedbitset::FixedBitSet;
+use std::marker::PhantomData;
+
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub struct Access<T: SparseSetIndex> {
+    reads_all: bool,
+    reads_and_writes: FixedBitSet,
+    writes: FixedBitSet,
+    marker: PhantomData<T>,
+}
+
+impl<T: SparseSetIndex> Default for Access<T> {
+    fn default() -> Self {
+        Self {
+            reads_all: false,
+            reads_and_writes: Default::default(),
```

Comment:
This isn't a very intuitive name

Reply:
Any suggestions? It is a combination of reads and writes. I don't have any better names than that

Reply:
Unless we go with something abbreviated which will almost surely be _less_ obvious, it seems like `reads_and_writes` is about as explanatory as it get's and I don't feel like it's too long.

Reply:
Hmm, in the context this feels like a list of the reads. `reads_and_writes` sounds like the accesses with mutable access. Having just a `reads` and `writes` (where the fact these also read is implicit) might be better.

Reply:
yeah thats a reasonable take. lets roll with reads() and writes()